crates/fluvio-cli/src/profile/sync/k8.rs

```diff
@@ -112,22 +112,22 @@ pub async fn set_k8_context(opt: K8Opt, external_addr: String) -> Result<Profile
 
 /// find fluvio addr
 pub async fn discover_fluvio_addr(namespace: Option<&str>) -> Result<Option<String>> {
-    use k8_client::http::status::StatusCode;
-
     let ns = namespace.unwrap_or("default");
     let svc = match K8Client::try_default()?
         .retrieve_item::<ServiceSpec, _>(&InputObjectMeta::named("fluvio-sc-public", ns))
         .await
     {
         Ok(svc) => svc,
-        Err(err) => match err {
-            k8_client::ClientError::ApiResponse(status)
-                if status.code == Some(StatusCode::NOT_FOUND.as_u16()) =>
+        Err(err) => {
+            if let Some(MetaStatus {
+                code: Some(404), ..
```

Comment:
use http crate?

Reply:
Not sure what you mean. 

This is just replacing the old

```
k8_client::ClientError::ApiResponse(status)
                if status.code == Some(StatusCode::NOT_FOUND.as_u16()) =>
```
                
using the anyhow downcasting to extract the status code from the inner error.